lib/web3-contracts.js

```diff
@@ -1,23 +1,27 @@
-import { useEffect, useRef, useState, useCallback } from 'react'
+import { useEffect, useRef, useState, useCallback, useMemo } from 'react'
 import { Contract as EthersContract } from 'ethers'
 import { getKnownContract } from './known-contracts'
-import tokenBalanceOfAbi from './token-balanceof.json'
 import { useWeb3Connect } from './web3-connect'
-import Web3EthContract from 'web3-eth-contract'
-import { calculateSlippageAmount, bigNum } from './utils'
+import { bigNum } from './utils'
 
-const NETWORK_AGENT_ADDR = '0x5E8c17A6065C35b172B10E80493D2266e2947DF4'
-const NETWORK_RESERVE_ADDR = '0xec0dd1579551964703246becfbf199c27cb84485'
+const CONNECTOR_WEIGHT = 250000
+const RETRY_EVERY = 1000
 
 const contractsCache = new Map()
+const tokenDecimals = new Map([
```

Comment:
To be really honest, I changed this to avoid a previous clash I was having with the contract cache, but this is no longer happening. I still prefer this implementation to save a few network calls. What do you guys think?

Reply:
Sounds good to me; we just need ANT and ANJ here and both of those we know can't have their token decimals changed.